src/regex/pcre2.cr

```diff
@@ -43,19 +43,25 @@ module Regex::PCRE2
 
   private def pcre2_options(options)
     flag = 0
-    options.each do |option|
-      flag |= case option
-              when .ignore_case?   then LibPCRE2::CASELESS
-              when .multiline?     then LibPCRE2::DOTALL | LibPCRE2::MULTILINE
-              when .extended?      then LibPCRE2::EXTENDED
-              when .anchored?      then LibPCRE2::ANCHORED
-              when .utf_8?         then LibPCRE2::UTF
-              when .no_utf8_check? then LibPCRE2::NO_UTF_CHECK
-              when .dupnames?      then LibPCRE2::DUPNAMES
-              when .ucp?           then LibPCRE2::UCP
-              else
-                raise "unreachable"
-              end
+    Regex::Options.each do |option|
+      if options.includes?(option)
+        flag |= case option
+                when .ignore_case?   then LibPCRE2::CASELESS
+                when .multiline?     then LibPCRE2::DOTALL | LibPCRE2::MULTILINE
+                when .extended?      then LibPCRE2::EXTENDED
+                when .anchored?      then LibPCRE2::ANCHORED
+                when .utf_8?         then LibPCRE2::UTF
+                when .no_utf8_check? then LibPCRE2::NO_UTF_CHECK
+                when .dupnames?      then LibPCRE2::DUPNAMES
+                when .ucp?           then LibPCRE2::UCP
+                else
+                  raise "unreachable"
+                end
+        options ^= option
```

Comment:
`Regex::Options.new(2)` would raise `Unknown Regex::Option value: 4`, whereas `Regex::Options.new(4)` would raise `Unknown Regex::Option value: 2`, even though they don't really use any unnamed members. Note that they both `includes?(Regex::Options::MULTILINE)`, hence the corresponding question methods return true. Maybe this is more appropriate:

```suggestion
        options &= ~option
```